Fixes not being able to put markers on monsters

Azmoria opened the pull request:
Fixes: #463

This bug only exists in main not the live version.

It also stops markers from going on player tokens if they don't have inspiration. This fixes that as well.

bain2236 commented:
I dropped the ball on this one. But this totally fixes it

vlaminck reviewed (changes requested):
`pop` doesn't work like that. I always pops no matter what the value is. I fixed this in your token context menu rebuild branch so this PR probably isn't needed any more, but this is how I fixed it


    if (this.options.inspiration){
        if (!this.options.custom_conditions.includes("Inspiration")){
            this.options.custom_conditions.push("Inspiration")
        }
    }
    else {
        array_remove_index_by_value(this.options.custom_conditions, "Inspiration");
    }

Azmoria commented:
> `pop` doesn't work like that. I always pops no matter what the value is. I fixed this in your token context menu rebuild branch so this PR probably isn't needed any more, but this is how I fixed it
> 
> ```
> if (this.options.inspiration){
>     if (!this.options.custom_conditions.includes("Inspiration")){
>         this.options.custom_conditions.push("Inspiration")
>     }
> }
> else {
>     array_remove_index_by_value(this.options.custom_conditions, "Inspiration");
> }
> ```

Yeh that makes sense I had just wrapped what was there to check if inspiration was there. This seemed to fix the issue so I left it. Closing this since your fix is in the other one
